src/main/java/uk/gov/hmcts/reform/sscs/ccd/presubmit/uploadhearingrecording/UploadHearingRecordingAboutToStartHandler.java

```diff
@@ -0,0 +1,90 @@
+import uk.gov.hmcts.reform.sscs.idam.IdamService;
+
+@Service
+@Slf4j
+public class UploadHearingRecordingAboutToStartHandler implements PreSubmitCallbackHandler<SscsCaseData> {
+
+    private final IdamService idamService;
+
+    @Autowired
+    public UploadHearingRecordingAboutToStartHandler(IdamService idamService) {
+        this.idamService = idamService;
+    }
+
+    @Override
+    public boolean canHandle(CallbackType callbackType, Callback<SscsCaseData> callback) {
+        requireNonNull(callback, "callback must not be null");
+        requireNonNull(callbackType, "callbacktype must not be null");
+
+        return callbackType.equals(CallbackType.ABOUT_TO_START)
+            && callback.getEvent() == EventType.UPLOAD_HEARING_RECORDING;
+    }
+
+    @Override
+    public PreSubmitCallbackResponse<SscsCaseData> handle(CallbackType callbackType, Callback<SscsCaseData> callback,
+                                                          String userAuthorisation) {
+        if (!canHandle(callbackType, callback)) {
+            throw new IllegalStateException("Cannot handle callback");
+        }
+
+        final CaseDetails<SscsCaseData> caseDetails = callback.getCaseDetails();
+        final SscsCaseData sscsCaseData = caseDetails.getCaseData();
+        PreSubmitCallbackResponse<SscsCaseData> response = new PreSubmitCallbackResponse<>(sscsCaseData);
+
+        if (sscsCaseData.getHearings() == null || sscsCaseData.getHearings().isEmpty()) {
+            response.addError("No hearing has been conducted on this case");
+            return response;
+        } else {
+            LocalDateTime currentTime = LocalDateTime.now();
+            List<DynamicListItem> validHearings = sscsCaseData.getHearings().stream()
+                .filter(hearing -> isBeforeDate(currentTime, hearing.getValue().getHearingDate(),
+                    hearing.getValue().getTime()))
+                .map(hearing -> new DynamicListItem(hearing.getValue().getHearingId(), selectHearing(hearing)))
+                .collect(Collectors.toList());
+            if (validHearings.isEmpty()) {
+                response.addError("No hearing has been conducted on this case");
+                return response;
+            }
+            sscsCaseData.setSelectHearingDetails(new DynamicList(new DynamicListItem("", ""), validHearings));
+        }
+        return response;
+    }
+
+    private boolean isBeforeDate(LocalDateTime currentDate, String hearingDate, String hearingTime) {
+        DateTimeFormatter formatter = DateTimeFormatter.ofPattern("yyyy-MM-dd HH:mm");
+        return LocalDateTime.parse(hearingDate + " " + hearingTime, formatter).isBefore(currentDate);
+    }
+
+    @NotNull
+    private String selectHearing(Hearing hearing) {
+        DateTimeFormatter formatter = DateTimeFormatter.ofPattern("yyyy-MM-dd");
+        DateTimeFormatter resultFormatter = DateTimeFormatter.ofPattern("dd MMM yyyy");
```

Comment:
these two variables could be private static final

Reply:
done